### PR TITLE
feat(ci): Check if PR is rebased on top of Master

### LINKED ISSUE #10648: Make sure master always pass unit tests
PR can be merged if all required unit tests are green but if the PR is not rebased, it can lead to failures in unit tests on master.
Solutions:
- Create a new required check that will block if the PR is not rebased.
- Track unit test in the new ci dashboard.

### PR DESCRIPTION
**Pull request by JeanPec**
Signed-off-by: JeanPec <andremarie67450@gmail.com>

## Summary
Added CI Job to test if PR is rebased on top of Master

## Test Plan
Tested both on local repo and magma/magma repo.
Tested when PR was and was not rebased.

## Additional Information
The Step returns True when the PR is rebased on top of master.
The Step returns False when the PR is not rebased on top of master.

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/check-rebase.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+    branches:
+      - master
+      - v1.*
+
+jobs:
+  checkRebase:
+    runs-on: ubuntu-latest
+    env:
+      PR_NUMBER: "${{ github.workspace }}"
```
Comment (quentinDERORY) [inline .github/workflows/check-rebase.yml]: remove this

File: .github/workflows/check-rebase.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/checkout@v2
+        with:
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
+      - name: Checkout Base
+        uses: actions/checkout@v2
+        with:
+          repository: ${{github.event.pull_request.base.repo.full_name}}
+          fetch-depth: 0
+      - run: |
```
Comment (quentinDERORY) [inline .github/workflows/check-rebase.yml]: define the head and base SHA as environment variable to make the code more readable

File: .github/workflows/check-rebase.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          BASE_HEAD_HASH=$(git merge-base ${{github.event.pull_request.base.sha}} ${{github.event.pull_request.head.sha}})
```
Comment (quentinDERORY) [inline .github/workflows/check-rebase.yml]: ReName BASE_HEAD_HASH to COMMON_SHA

File: .github/workflows/check-rebase.yml (hunk 2 above)
Comment (quentinDERORY) [inline .github/workflows/check-rebase.yml]: Name the step check commits hash

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
See comments

**Review by quentinDERORY — APPROVED** [review]

**Review by tmdzk — APPROVED** [review]

**Pull request merged**